packages/ez-react/src/components/StackedBars.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, SVGAttributes, useMemo } from 'react';
+import { scaleRectangleData } from 'eazychart-core/src';
+import { Bar } from '@/components/shapes/Bar';
+import { useChart } from '@/lib/use-chart';
+import { useCartesianScales } from '@/components/scales/CartesianScale';
+import { useColorScale } from './scales/ColorScale';
+import { RectangleDatum } from 'eazychart-core/src/types';
+
+export interface StackedBarsProps extends SVGAttributes<SVGGElement> {
+  xDomainKey: string;
+  yDomainKeys: string[];
+}
+
+export const StackedBars: FC<StackedBarsProps> = ({
+  xDomainKey,
+  yDomainKeys,
+  ...rest
+}) => {
+  const { data, dimensions, isRTL } = useChart();
+  const { xScale, yScale } = useCartesianScales();
+  const { colorScale } = useColorScale();
+
+  const scaledData = useMemo(() => {
+    return yDomainKeys.reduce((acc, yDomainKey, index) => {
+      if (index === 0) {
+        // @ts-ignore
+        acc[yDomainKey] = scaleRectangleData(
+          data,
+          xDomainKey,
+          yDomainKey,
+          xScale,
+          yScale,
+          colorScale,
+          dimensions,
+          isRTL
+        );
+        return acc;
+      } else {
+        // @ts-ignore
+        acc[yDomainKey] = scaleRectangleData(
+          data,
+          xDomainKey,
+          yDomainKey,
+          xScale,
+          yScale,
+          colorScale,
+          dimensions,
+          isRTL
+        ).map((datum, id) => {
```

Comment:
I would suggest that we create a another scale function called "scaleStackedRectangleData()" wdyt ?

Reply:
I think it will be a lot cleaner